fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/GraphExpansion.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * GraphExpansion.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.temp;
+
+import com.apple.foundationdb.record.query.plan.temp.expressions.SelectExpression;
+import com.apple.foundationdb.record.query.predicates.AndPredicate;
+import com.apple.foundationdb.record.query.predicates.QuantifiedColumnValue;
+import com.apple.foundationdb.record.query.predicates.QueryPredicate;
+import com.apple.foundationdb.record.query.predicates.Value;
+import com.apple.foundationdb.record.query.predicates.ValueComparisonRangePredicate;
+import com.apple.foundationdb.record.query.predicates.ValueComparisonRangePredicate.Placeholder;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Class to abstract behavior when query expansion is applied to query components or key expressions. An object of this
+ * class represents a conglomeration for result values, predicates, quantifiers before those elements are unified and
+ * eventually find their homw in a {@link SelectExpression}.
+ *
+ * Another way of thinking about this class is to conceptualize logic implemented her as a pre-stage to actually
+ * creating expressions, as a builder of a {@link SelectExpression}s of sorts. The fundamental difference between
+ * a builder in an ordinary builder pattern and this class also provides getters to all elements and the ability to
+ * merge other objects of this class in a additive manner.
```

Comment:
"her" -> "here" But, really, both sentences in this paragraph could use some wordsmithing.